src/logic_checker.rb

```diff
@@ -0,0 +1,79 @@
+require './ranges'
+
+class LogicChecker
+	NUM_TYPES = ["intField", "signedIntField", "unsignedIntField", "signedShortField",
+		"unsignedShortField", "signedByteField", "unsignedByteField", "booleanField"]
+
+	def checkLogic(model)
+		model.field.each do |field|
+			if NUM_TYPES.include? field.type
+				checkNumFieldLogic(field, false)
+			elsif field.type === "doubleField" or field.type === "floatField"
+				checkNumFieldLogic(field, true)
+			elsif field.type === "stringField"
+				checkStringField(field)
+			end
+		end
+		model.group.each do |group|
+			checkLogic(group)
+		end	
+	end
+
+	def checkNumFieldLogic(field, decimal)
+		absoluteMin = Object.const_get(field.type.upcase + "_MIN")
+		absoluteMax = Object.const_get(field.type.upcase + "_MAX")
+		
+		!decimal ? value = field.value.chardata[0].to_i : value = field.value.chardata[0].to_f
+		
+		if defined? field.min and not field.min.empty?
+			!decimal ? min = field.min.chardata[0].strip.to_i : min = field.min.chardata[0].strip.to_f
+			if min < absoluteMin or value < min
+				puts "\nERROR: check min #{field.title.chardata[0].strip} value in XML"
```

Comment:
Would it be possible to make the error messages more verbose/clearer? For example:
`ERROR: check "min" (25) and "value" (20) in "A number", values are inconsistent!`